packages/fuels-core/src/types/wrappers/coin_type.rs

```diff
@@ -47,10 +44,10 @@ impl CoinType {
         }
     }
 
-    pub fn asset_id(&self) -> AssetId {
+    pub fn asset_id(&self) -> Option<AssetId> {
```

Comment:
maybe call it `coin_asset_id`. it being the base asset isn't clear from returning `None`.

Or remove the helper altogether.